joschahenningsen/TUM-Live
# Multiple Synced Players

### Motivation and Context
Adds multiple synced players as a viewing option. The current split-view implementation is very primitive and could
be enhanced in a later PR. An example off the top of my head: Show the chat below the CAM version for a bigger PRES stream. 

 ### Description
Adds View Version `STEP` with a ratio-picker with predefined ratios (3:1, 1:1, 1:3) and a splitter (split.js) for a fully customized view.

 ### Steps for Testing
 1. Navigate to a livestream which has a `PRES` and a `CAM` version
 2. Click on the split-view button in the actions-bar. You will be redirect to `.../SPLIT`
 3. Hovering over the video makes the ratio-picker visible
 4. Choose a ratio. Controls are only visible for "focused" player.
 6. Make sure to test `CAM`, `PRES`, and `COMB` on its own
 7. Make sure to test the chat replay

 ### Screenshots

#### View selector

<div style='display:flex'>
<img width="301" alt="image" src="https://user-images.githubusercontent.com/29633518/200484073-a8e96a71-9c97-4086-a6ff-bc3cab3a1ab9.png">

<img width="300" alt="image" src="https://user-images.githubusercontent.com/29633518/200484178-c614aa8f-e33c-43e7-afea-8663b952a9a7.png">
</div>

#### Split menu


https://user-images.githubusercontent.com/29633518/200484671-424ceacc-18c8-4c3d-88c1-208cc970acd4.mov
